integrationTests/state/stateTrie_test.go

```diff
@@ -1025,6 +1032,152 @@ func TestAccountsDB_ExecALotOfBalanceTxOKorNOK(t *testing.T) {
 	adbPrintAccount(acntDest.(*state.Account), "Destination")
 }
 
+func BenchmarkCreateOneMillionAccountsWithMockDB(b *testing.B) {
+	nrOfAccounts := 1000000
+	nrTxs := 15000
+	txVal := 100
+	balance := nrTxs * txVal
+	persist := mock2.MockDB{}
+
+	adb, _ := createAccounts(b, nrOfAccounts, balance, persist)
+	var rtm runtime.MemStats
+	runtime.GC()
+
+	runtime.ReadMemStats(&rtm)
+	fmt.Printf("Mem before committing %v accounts with mockDB: go mem - %s, sys mem - %s \n",
+		nrOfAccounts,
+		core.ConvertBytes(rtm.Alloc),
+		core.ConvertBytes(rtm.Sys),
+	)
+
+	_, _ = adb.Commit()
+
+	runtime.ReadMemStats(&rtm)
+	fmt.Printf("Mem after committing %v accounts with mockDB: go mem - %s, sys mem - %s \n",
+		nrOfAccounts,
+		core.ConvertBytes(rtm.Alloc),
+		core.ConvertBytes(rtm.Sys),
+	)
```

Comment:
Maybe add a new runtime.GC() and a new print here.
Also, after the last GC, try to iterate the trie (maybe with the use of trie.String()) and check if all nodes are iterated. If 1 million accounts is too much, try a smaller test with 10 accounts (in this test mem usage will not matter, just to be sure the nodes exist)

Reply:
Done.